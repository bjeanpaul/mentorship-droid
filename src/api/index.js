export * from 'src/api/schemas';
export * from 'src/api/profiles';
export * from 'src/api/schedule';
export * from 'src/api/activities';
export * from 'src/api/categories';
export * from 'src/api/events';
export * from 'src/api/errors';
export * from 'src/api/sync';
export * from 'src/api/calls';
export * from 'src/api/notificationSettings';
export * from 'src/api/callNotes';
export * from 'src/api/messages';
export * from 'src/api/imageUrl';
export { default as imageUrl } from 'src/api/imageUrl';
