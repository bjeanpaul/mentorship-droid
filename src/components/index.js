import Text from './Text';
import Label from './Label';
import Link from './Link';
import Button from './Button';
import TextInput from './TextInput';
import MultiLineTextInput from './MultiLineTextInput';
import BaseView from './BaseView';
import Toolbar from './Toolbar';
import Header from './Header';
import Title from './Title';
import Icon from './Icon';
import HeaderIcon from './HeaderIcon';
import OverlayCompleted from './Overlay/Completed';
import OverlayLoading from './Overlay/Loading';
import Section from './Section';

export {
  BaseView,
  Toolbar,
  Text,
  Label,
  Link,
  Button,
  TextInput,
  MultiLineTextInput,
  Toolbar,
  Header,
  Title,
  Icon,
  HeaderIcon,
  OverlayCompleted,
  OverlayLoading,
  Section,
};
