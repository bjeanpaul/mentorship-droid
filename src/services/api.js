const HOSTNAME = 'http://localhost:8000/mentor'

export function callAPI(endpoint, hash) {

  return fetch(`${HOSTNAME}${endpoint}`, {
    headers: {
      'Authorization': `Basic ${hash}`
    }
  })
  .then(response => response.json().then(json => ({ json, response })))
  .then(({json, response}) => {
    if (response.ok === false) {
      return Promise.reject(json)
    }
    return json
  })

}
