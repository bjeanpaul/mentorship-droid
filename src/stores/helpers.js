import moment from 'moment';
import { values, isUndefined, sortBy, filter, first } from 'lodash';


export const getAuthUserProfile = ({
  auth: { profileId },
  entities: { profiles },
}) => !isUndefined(profileId)
  ? profiles[profileId]
  : null;


export const getContext = state => {
  const { auth: { auth } } = state;

  return {
    auth,
    profile: getAuthUserProfile(state),
  };
};


// TODO sort?
export const getCategories = ({ entities: { categories } }) => values(categories);


// TODO sort?
export const getCategoryActivities = ({
  entities: { activities },
}, targetCategoryId) => (
  values(activities)
    .filter(({ category }) => category === targetCategoryId));


export const getCategory = ({ entities: { categories } }, id) => categories[id];


export const getActivity = ({ entities: { activities } }, id) => activities[id];


export const getScheduledCalls = ({ entities: { scheduledCalls } }) => values(scheduledCalls);


export const getScheduledCall = ({ entities: { scheduledCalls } }, id) => scheduledCalls[id];


export const getScheduledCallActivity = (state, id) => {
  const scheduledCall = getScheduledCall(state, id);
  return scheduledCall && scheduledCall.activity && getActivity(state, scheduledCall.activity);
};


export const getEvents = ({ entities: { events } }) => values(events);


export const getActivityCallNotes = ({
  entities: { callNotes },
}, activityId) => (
  filter(callNotes, { callActivity: activityId }));


export const getCall = ({ entities: { calls } }, id) => calls[id];


export const getCallNote = ({ entities: { callNotes } }, id) => callNotes[id];


export const getNextScheduledCall = (state, predicate = {}, time = Date.now()) => {
  let res = sortBy(getScheduledCalls(state), ({ callTime }) => callTime);
  res = res.filter(({ callTime }) => moment(callTime).isSameOrAfter(time));
  res = filter(res, predicate);
  return first(res);
};
