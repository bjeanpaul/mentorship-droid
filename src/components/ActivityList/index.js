import React, { PropTypes } from 'react';
import { View, Image, ScrollView, TouchableNativeFeedback } from 'react-native';
import { Text } from 'src/components';
import images from 'src/constants/images';
import { ImageUrl } from 'src/api';
import styles from './styles';


const ActivityList = ({
  category: { color },
  activities,
  onActivityPress,
}) => (
  <ScrollView>
    {activities.map(({ id, title, icon, isComplete }) => (
    <TouchableNativeFeedback
      key={id}
      uid={`activity:${id}`}
      onPress={() => onActivityPress(id)}
    >
      <View style={styles.activity}>
        <ActivityIcon icon={icon} color={color} isComplete={isComplete} />
        <Text numberOfLines={3} style={styles.activityTitle}>{title}</Text>
      </View>
    </TouchableNativeFeedback>
    ))}
  </ScrollView>
);


const ActivityIcon = ({
  icon,
  color,
  isComplete,
}) => (
  <View style={styles.activityIconContainer}>
    <View
      style={[
        styles.activityIcon,
        isComplete
          ? { backgroundColor: color }
          : styles.activityIconIsIncomplete,
      ]}
    >
      <Image
        source={icon.toSource(images.ACTIVITY_ICON_FALLBACK)}
        style={styles.activityIconImage}
      />
    </View>

    {
      isComplete
        ? <Image
          style={styles.activityCompleteAnnotation}
          source={images.ACTIVITY_COMPLETE_ANNOTATION}
        />
        : null
    }
  </View>
);


ActivityList.propTypes = {
  category: PropTypes.object.isRequired,
  activities: PropTypes.array.isRequired,
  onActivityPress: PropTypes.func.isRequired,
};


ActivityIcon.propTypes = {
  icon: PropTypes.instanceOf(ImageUrl),
  color: PropTypes.string.isRequired,
  isComplete: PropTypes.bool.isRequired,
};


export default ActivityList;
