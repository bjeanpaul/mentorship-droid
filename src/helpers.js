import { find, isUndefined, partialRight as partial } from 'lodash';
import { getBaseURL } from 'src/configuration';
import { omitBy, isNull } from 'lodash';


export const omitNulls = d => omitBy(d, isNull);


export const trap = (type, fn) => v => v instanceof type
  ? Promise.resolve(v).then(fn)
  : Promise.reject(v);


export const switchError = cases => (e, ...xargs) => {
  const [type, fn] = find(cases, ([type, fn]) => e instanceof type) || [];
  return !isUndefined(fn)
    ? Promise.resolve().then(() => fn(e, ...xargs))
    : Promise.reject(e);
};


export const apiAction = ({
  method,
  request,
  success,
  failures,
}) => (...args) => (dispatch, { auth }) => Promise.resolve()
  .then(() => request(...args))
  .then(dispatch)
  .then(() => method(...args, auth))
  .then(partial(success, ...args), partial(switchError(failures), ...args))
  .then(dispatch);


export const staticAction = type => () => ({
  type
});


export const dataAction = type => data => ({
  type,
  payload: data
});


export const generateActionTypes = function generateActionNames(
  resource,
  operations = ['FETCH', 'CREATE', 'UPDATE', 'DELETE']
) {
  const actionTypes = {};
  const resourceUC = resource.toUpperCase();

  operations.forEach((operation) => {
    const operationUC = operation.toUpperCase();
    const request = `${resourceUC}_${operationUC}_REQUEST`;
    const success = `${resourceUC}_${operationUC}_SUCCESS`;
    const failure = `${resourceUC}_${operationUC}_FAILURE`;

    const operationLC = operation.toLowerCase();
    actionTypes[`${operationLC}Request`] = request;
    actionTypes[`${operationLC}Success`] = success;
    actionTypes[`${operationLC}Failure`] = failure;
  });
  return actionTypes;
};


export const filterActionTypes = function filterActionTypes(
  actionTypes,
  operation
) {
  const [request, success, failure] = [
    `${operation}Request`,
    `${operation}Success`,
    `${operation}Failure`,
  ];

  if (actionTypes[request] || actionTypes[success] || actionTypes[failure]) {
    return [
      actionTypes[request],
      actionTypes[success],
      actionTypes[failure],
    ];
  }
  return [];
};


// TODO: Delete
export const generateActionCreators = function generateActionCreators({
  resourcePath,
  actionTypes,
  requestOpts = {},
  normalizeJSON,
}) {
  const actionCreators = {};
  let url = `${getBaseURL()}/`;
  if (resourcePath) {
    url += `${resourcePath}/`;
  }

  const fetchActionTypes = filterActionTypes(actionTypes, 'fetch');
  if (fetchActionTypes.length === 3) {
    actionCreators.fetch = (onSuccess, onFailure) => ({
      type: '--generated fetch--',
      types: fetchActionTypes,
      url,
      requestOpts,
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }

  const createActionTypes = filterActionTypes(actionTypes, 'create');
  if (createActionTypes.length === 3) {
    actionCreators.create = ({
      body,
      onSuccess,
      onFailure,
    }) => ({
      type: '--generated create--',
      types: createActionTypes,
      url,
      requestOpts: { ...requestOpts,
        method: 'POST',
        body: JSON.stringify(body),
      },
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }

  // id is required as an argument to update a record;
  // we have a scenario with profiless, where it's
  // profile/{id}/image,
  // which is awkward...
  const updateActionTypes = filterActionTypes(actionTypes, 'update');
  if (updateActionTypes.length === 3) {
    actionCreators.update = ({
      id,
      body,
      onSuccess,
      onFailure,
    }) => ({
      type: '--generated update--',
      types: updateActionTypes,
      url: `${url}${id}/`,
      requestOpts: { ...requestOpts,
        method: 'PUT',
        body: JSON.stringify(body),
      },
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }


  return actionCreators;
};
