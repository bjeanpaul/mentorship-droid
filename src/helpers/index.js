import { getBaseURL } from 'src/configuration';
import { omitBy, isNull } from 'lodash';


export const conj = (...args) => Object.assign({}, ...args);


export const omitNulls = d => omitBy(d, isNull);


export const generateActionTypes = function generateActionNames(
  resource,
  operations = ['FETCH', 'CREATE', 'UPDATE', 'DELETE']
) {
  const actionTypes = {};
  const resourceUC = resource.toUpperCase();

  operations.forEach((operation) => {
    const operationUC = operation.toUpperCase();
    const request = `${resourceUC}_${operationUC}_REQUEST`;
    const success = `${resourceUC}_${operationUC}_SUCCESS`;
    const failure = `${resourceUC}_${operationUC}_FAILURE`;

    const operationLC = operation.toLowerCase();
    actionTypes[`${operationLC}Request`] = request;
    actionTypes[`${operationLC}Success`] = success;
    actionTypes[`${operationLC}Failure`] = failure;
  });
  return actionTypes;
};


export const filterActionTypes = function filterActionTypes(
  actionTypes,
  operation
) {
  const [request, success, failure] = [
    `${operation}Request`,
    `${operation}Success`,
    `${operation}Failure`,
  ];

  if (actionTypes[request] || actionTypes[success] || actionTypes[failure]) {
    return [
      actionTypes[request],
      actionTypes[success],
      actionTypes[failure],
    ];
  }
  return [];
};


// TODO: Delete
export const generateActionCreators = function generateActionCreators({
  resourcePath,
  actionTypes,
  requestOpts = {},
  normalizeJSON,
}) {
  const actionCreators = {};
  let url = `${getBaseURL()}/`;
  if (resourcePath) {
    url += `${resourcePath}/`;
  }

  const fetchActionTypes = filterActionTypes(actionTypes, 'fetch');
  if (fetchActionTypes.length === 3) {
    actionCreators.fetch = (onSuccess, onFailure) => ({
      type: '--generated fetch--',
      types: fetchActionTypes,
      url,
      requestOpts,
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }

  const createActionTypes = filterActionTypes(actionTypes, 'create');
  if (createActionTypes.length === 3) {
    actionCreators.create = ({
      body,
      onSuccess,
      onFailure,
    }) => ({
      type: '--generated create--',
      types: createActionTypes,
      url,
      requestOpts: { ...requestOpts,
        method: 'POST',
        body: JSON.stringify(body),
      },
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }

  // id is required as an argument to update a record;
  // we have a scenario with profiless, where it's
  // profile/{id}/image,
  // which is awkward...
  const updateActionTypes = filterActionTypes(actionTypes, 'update');
  if (updateActionTypes.length === 3) {
    actionCreators.update = ({
      id,
      body,
      onSuccess,
      onFailure,
    }) => ({
      type: '--generated update--',
      types: updateActionTypes,
      url: `${url}${id}/`,
      requestOpts: { ...requestOpts,
        method: 'PUT',
        body: JSON.stringify(body),
      },
      normalizeJSON,
      onSuccess,
      onFailure,
    });
  }


  return actionCreators;
};
