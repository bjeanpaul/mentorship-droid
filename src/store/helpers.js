import moment from 'moment';
import { values, isUndefined, sortBy, filter, first, orderBy } from 'lodash';

import { getCurrent } from 'src/navigationHelpers';
import { TYPES_TO_COLLECTIONS } from 'src/constants/events';


export const getAuthUserProfile = ({
  auth: { profileId },
  entities: { profiles },
}) => !isUndefined(profileId)
  ? profiles[profileId]
  : null;


export const getContext = state => {
  const { auth: { auth } } = state;

  return {
    auth,
    profile: getAuthUserProfile(state),
  };
};


export const getCategories = ({ entities }) =>
  sortBy(values(entities.categories), 'ordinal');


export const getCategoryActivities = ({ entities }, targetCategoryId) => {
  const activities = values(entities.activities)
    .filter(({ category }) => category === targetCategoryId);

  return sortBy(activities, 'ordinal');
};


export const getCategory = ({ entities: { categories } }, id) =>
  categories[id];


export const getActivity = ({ entities: { activities } }, id) =>
  activities[id];


export const getScheduledCalls = ({ entities: { scheduledCalls } }) =>
  values(scheduledCalls);


export const getScheduledCall = ({ entities: { scheduledCalls } }, id) =>
  scheduledCalls[id];


export const getScheduledCallActivity = (state, id) => {
  const scheduledCall = getScheduledCall(state, id);
  return scheduledCall
      && scheduledCall.activity
      && getActivity(state, scheduledCall.activity);
};


export const getActivityCallNotes = ({
  entities: { callNotes },
}, activityId) =>
  values(callNotes)
    .filter(({ callActivityId }) => callActivityId === activityId);


export const getCallNotes = ({
  entities: { callNotes },
}) =>
  sortBy(callNotes, ({ callStartTime }) => +moment(callStartTime));


export const getCall = ({ entities: { calls } }, id) => calls[id];


export const getCallNote = ({ entities: { callNotes } }, id) => callNotes[id];


export const getNextScheduledCall = (state, predicate = {}, time = Date.now()) => {
  let res = sortBy(getScheduledCalls(state), ({ callTime }) => +moment(callTime));
  res = res.filter(({ callTime }) => moment(callTime).isSameOrAfter(time));
  res = filter(res, predicate);
  return first(res);
};


export const getScheduledCallsBetween = (state, start, end, time = Date.now()) => {
  // for some reason, moment durations are behaving inconsistently in tests,
  // thus the lack of moment usage here
  const startTime = +moment(time) - start;
  const endTime = +moment(time) + end;

  return getScheduledCalls(state)
    .filter(({ callTime }) => {
      const t = +moment(callTime);
      return startTime <= t && t <= endTime;
    });
};


const getEventObject = (state, event) => {
  const collection = state.entities[TYPES_TO_COLLECTIONS[event.eventType]];
  return collection && collection[event.objectId];
};


// TODO return the associated entitites to make the containers' jobs easier
export const getEvents = (state) =>
  values(state.entities.events)
    .filter(event => getEventObject(state, event));


export const getMessages = ({ entities }) => []
  .concat(values(entities.messages))
  .concat(values(entities.pendingMessages));


export const getMessage = ({ entities }, id) => entities.messages[id];


export const getActiveTopRoute = store => getCurrent(store.navigation.top);


export const getBlogPost = ({ entities }, id) => entities.blogPosts[id];


export const getBlogPosts = ({ entities }) =>
  orderBy(entities.blogPosts, [({ createdAt }) => +moment(createdAt)], ['desc']);
