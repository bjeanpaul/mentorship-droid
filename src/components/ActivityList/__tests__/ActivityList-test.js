import React from 'react';
import { noop } from 'lodash';
import ActivityList from 'src/components/ActivityList';
import { fakeCategory, fakeActivity, uidEquals } from 'app/scripts/helpers';
import { imageUrl } from 'src/api';


describe('ActivityList', () => {
  function createComponent(props = {}) {
    return (
      <ActivityList
        onActivityPress={noop}
        category={fakeCategory()}
        activities={[{
          id: 1,
          title: 'Activity 1',
        }, {
          id: 2,
          title: 'Activity 2',
        }].map(fakeActivity)}
        {...props}
      />
    );
  }

  it('should render', () => {
    expect(render(createComponent())).toMatchSnapshot();
  });

  it('should render incomplete activities', () => {
    expect(render(createComponent({
      activities: [
        fakeActivity({
          icon: imageUrl('/foo.jpg'),
          isComplete: false,
        }),
      ],
    }))).toMatchSnapshot();
  });

  it('should render complete activities', () => {
    expect(render(createComponent({
      activities: [
        fakeActivity({
          icon: imageUrl('/foo.jpg'),
          isComplete: true,
        }),
      ],
    }))).toMatchSnapshot();
  });

  it('should render a fallback for incomplete activities without an icon', () => {
    expect(render(createComponent({
      activities: [
        fakeActivity({
          icon: null,
          isComplete: false,
        }),
      ],
    }))).toMatchSnapshot();
  });

  it('should render a fallback for complete activities without an icon', () => {
    expect(render(createComponent({
      activities: [
        fakeActivity({
          icon: null,
          isComplete: true,
        }),
      ],
    }))).toMatchSnapshot();
  });

  it('should call onActivityPress when a activity is pressed', () => {
    const onActivityPress = jest.fn();
    const el = shallow(createComponent({ onActivityPress }));

    el.findWhere(uidEquals('activity:1'))
      .simulate('press');

    expect(onActivityPress.mock.calls).toEqual([[1]]);

    el.findWhere(uidEquals('activity:2'))
      .simulate('press');

    expect(onActivityPress.mock.calls).toEqual([[1], [2]]);
  });
});
