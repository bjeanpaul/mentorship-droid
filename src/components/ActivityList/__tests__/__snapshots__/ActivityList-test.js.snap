exports[`ActivityList should render 1`] = `
<ScrollView>
  <TouchableNativeFeedback
    onPress={[Function]}
    uid="activity:1">
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderBottomColor": "#dfe5e6",
          "borderBottomWidth": 1,
          "height": 120,
          "padding": 16,
        }
      }>
      <View
        style={
          Object {
            "position": "absolute",
            "top": 24,
          }
        }>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 72,
                "height": 72,
                "justifyContent": "center",
                "width": 72,
              },
              Object {
                "backgroundColor": "#ebebeb",
              },
            ]
          }>
          <Image
            source={1}
            style={
              Object {
                "borderRadius": 72,
                "height": 72,
                "width": 72,
              }
            } />
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={3}
        style={
          Array [
            Object {
              "color": "#003035",
              "fontFamily": "Brandon Text",
            },
            Object {
              "fontSize": 13,
              "textAlign": "center",
            },
            Object {
              "fontSize": 18,
              "marginLeft": 104,
              "textAlign": "left",
            },
          ]
        }>
        Activity 1
      </Text>
    </View>
  </TouchableNativeFeedback>
  <TouchableNativeFeedback
    onPress={[Function]}
    uid="activity:2">
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderBottomColor": "#dfe5e6",
          "borderBottomWidth": 1,
          "height": 120,
          "padding": 16,
        }
      }>
      <View
        style={
          Object {
            "position": "absolute",
            "top": 24,
          }
        }>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 72,
                "height": 72,
                "justifyContent": "center",
                "width": 72,
              },
              Object {
                "backgroundColor": "#ebebeb",
              },
            ]
          }>
          <Image
            source={1}
            style={
              Object {
                "borderRadius": 72,
                "height": 72,
                "width": 72,
              }
            } />
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={3}
        style={
          Array [
            Object {
              "color": "#003035",
              "fontFamily": "Brandon Text",
            },
            Object {
              "fontSize": 13,
              "textAlign": "center",
            },
            Object {
              "fontSize": 18,
              "marginLeft": 104,
              "textAlign": "left",
            },
          ]
        }>
        Activity 2
      </Text>
    </View>
  </TouchableNativeFeedback>
</ScrollView>
`;

exports[`ActivityList should render a fallback for complete activities without an icon 1`] = `
<ScrollView>
  <TouchableNativeFeedback
    onPress={[Function]}
    uid="activity:1">
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderBottomColor": "#dfe5e6",
          "borderBottomWidth": 1,
          "height": 120,
          "padding": 16,
        }
      }>
      <View
        style={
          Object {
            "position": "absolute",
            "top": 24,
          }
        }>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 72,
                "height": 72,
                "justifyContent": "center",
                "width": 72,
              },
              Object {
                "backgroundColor": "#97c13c",
              },
            ]
          }>
          <Image
            source={1}
            style={
              Object {
                "borderRadius": 72,
                "height": 72,
                "width": 72,
              }
            } />
        </View>
        <Image
          source={1}
          style={
            Object {
              "height": 24,
              "left": 52,
              "position": "absolute",
              "top": -4,
              "width": 24,
            }
          } />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={3}
        style={
          Array [
            Object {
              "color": "#003035",
              "fontFamily": "Brandon Text",
            },
            Object {
              "fontSize": 13,
              "textAlign": "center",
            },
            Object {
              "fontSize": 18,
              "marginLeft": 104,
              "textAlign": "left",
            },
          ]
        }>
        Activity 3
      </Text>
    </View>
  </TouchableNativeFeedback>
</ScrollView>
`;

exports[`ActivityList should render a fallback for incomplete activities without an icon 1`] = `
<ScrollView>
  <TouchableNativeFeedback
    onPress={[Function]}
    uid="activity:1">
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderBottomColor": "#dfe5e6",
          "borderBottomWidth": 1,
          "height": 120,
          "padding": 16,
        }
      }>
      <View
        style={
          Object {
            "position": "absolute",
            "top": 24,
          }
        }>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 72,
                "height": 72,
                "justifyContent": "center",
                "width": 72,
              },
              Object {
                "backgroundColor": "#ebebeb",
              },
            ]
          }>
          <Image
            source={1}
            style={
              Object {
                "borderRadius": 72,
                "height": 72,
                "width": 72,
              }
            } />
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={3}
        style={
          Array [
            Object {
              "color": "#003035",
              "fontFamily": "Brandon Text",
            },
            Object {
              "fontSize": 13,
              "textAlign": "center",
            },
            Object {
              "fontSize": 18,
              "marginLeft": 104,
              "textAlign": "left",
            },
          ]
        }>
        Activity 3
      </Text>
    </View>
  </TouchableNativeFeedback>
</ScrollView>
`;

exports[`ActivityList should render complete activities 1`] = `
<ScrollView>
  <TouchableNativeFeedback
    onPress={[Function]}
    uid="activity:1">
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderBottomColor": "#dfe5e6",
          "borderBottomWidth": 1,
          "height": 120,
          "padding": 16,
        }
      }>
      <View
        style={
          Object {
            "position": "absolute",
            "top": 24,
          }
        }>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 72,
                "height": 72,
                "justifyContent": "center",
                "width": 72,
              },
              Object {
                "backgroundColor": "#97c13c",
              },
            ]
          }>
          <Image
            source={
              Object {
                "uri": "/api/foo.jpg?filter_spec=min-144x144",
              }
            }
            style={
              Object {
                "borderRadius": 72,
                "height": 72,
                "width": 72,
              }
            } />
        </View>
        <Image
          source={1}
          style={
            Object {
              "height": 24,
              "left": 52,
              "position": "absolute",
              "top": -4,
              "width": 24,
            }
          } />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={3}
        style={
          Array [
            Object {
              "color": "#003035",
              "fontFamily": "Brandon Text",
            },
            Object {
              "fontSize": 13,
              "textAlign": "center",
            },
            Object {
              "fontSize": 18,
              "marginLeft": 104,
              "textAlign": "left",
            },
          ]
        }>
        Activity 3
      </Text>
    </View>
  </TouchableNativeFeedback>
</ScrollView>
`;

exports[`ActivityList should render incomplete activities 1`] = `
<ScrollView>
  <TouchableNativeFeedback
    onPress={[Function]}
    uid="activity:1">
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderBottomColor": "#dfe5e6",
          "borderBottomWidth": 1,
          "height": 120,
          "padding": 16,
        }
      }>
      <View
        style={
          Object {
            "position": "absolute",
            "top": 24,
          }
        }>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 72,
                "height": 72,
                "justifyContent": "center",
                "width": 72,
              },
              Object {
                "backgroundColor": "#ebebeb",
              },
            ]
          }>
          <Image
            source={
              Object {
                "uri": "/api/foo.jpg?filter_spec=min-144x144",
              }
            }
            style={
              Object {
                "borderRadius": 72,
                "height": 72,
                "width": 72,
              }
            } />
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={3}
        style={
          Array [
            Object {
              "color": "#003035",
              "fontFamily": "Brandon Text",
            },
            Object {
              "fontSize": 13,
              "textAlign": "center",
            },
            Object {
              "fontSize": 18,
              "marginLeft": 104,
              "textAlign": "left",
            },
          ]
        }>
        Activity 3
      </Text>
    </View>
  </TouchableNativeFeedback>
</ScrollView>
`;
